packages/esm-patient-clinical-view-app/src/defaulter-tracing/defaulter-tracing.component.tsx

```diff
@@ -24,105 +41,164 @@ interface PatientTracingProps {
-            : formatDate(parseDate(getObsFromEncounter(encounter, MissedAppointmentDate_UUID)));
-        },
-      },
-      {
-        key: 'visitDate',
-        header: t('visitDate', 'Tracing Date'),
-        getValue: (encounter) => {
-          return formatDate(parseDate(encounter.encounterDatetime));
-        },
-      },
-      {
-        key: 'tracingType',
-        header: t('tracingType', 'Tracing Type'),
-        getValue: (encounter) => {
-          return getObsFromEncounter(encounter, TracingType_UUID);
-        },
-      },
-      {
-        key: 'tracingNumber',
-        header: t('tracingNumber', 'Tracing No.'),
-        getValue: (encounter) => {
-          return getObsFromEncounter(encounter, TracingNumber_UUID);
-        },
-      },
-      {
-        key: 'contacted',
-        header: t('contacted', 'Contacted'),
-        getValue: (encounter) => {
-          return getObsFromEncounter(encounter, Contacted_UUID);
-        },
-      },
-      {
-        key: 'finalOutcome',
-        header: t('finalOutcome', 'Final Outcome'),
-        getValue: (encounter) => {
-          return getObsFromEncounter(encounter, TracingOutcome_UUID);
-        },
-      },
-      {
-        key: 'actions',
-        header: t('actions', 'Actions'),
-        getValue: (encounter) => {
-          const baseActions = [
-            {
-              form: { name: PatientTracingFormName, package: 'hiv' },
-              encounterUuid: encounter.uuid,
-              intent: '*',
-              label: 'Edit Form',
-              mode: 'edit',
-            },
-          ];
-          return baseActions;
-        },
-      },
-    ],
-    [],
-  );
+  const tableHeader = [
+    {
+      key: 'missedAppointmentDate',
+      header: t('missedAppointmentDate', 'Date Missed Appointment'),
+    },
+    {
+      key: 'visitDate',
+      header: t('visitDate', 'Tracing Date'),
+    },
+    {
+      key: 'tracingType',
+      header: t('tracingType', 'Tracing Type'),
+    },
+    {
+      key: 'tracingNumber',
+      header: t('tracingNumber', 'Tracing No.'),
+    },
+    {
+      key: 'contacted',
+      header: t('contacted', 'Contacted'),
+    },
+    {
+      key: 'finalOutcome',
+      header: t('finalOutcome', 'Final Outcome'),
+    },
+    {
+      key: 'actions',
+      header: t('actions', 'Actions'),
+    },
+  ];
+
+  const tableRows = encounters.map((encounter, index) => {
+    return {
+      id: `${encounter.uuid}`,
+      missedAppointmentDate:
+        getObsFromEncounter(encounter, MissedAppointmentDate_UUID) == '--' ||
+        getObsFromEncounter(encounter, MissedAppointmentDate_UUID) == null
+          ? formatDate(parseDate(encounter.encounterDatetime))
+          : formatDate(parseDate(getObsFromEncounter(encounter, MissedAppointmentDate_UUID))),
+      visitDate: formatDate(new Date(encounter.encounterDatetime)),
+      tracingType: getObsFromEncounter(encounter, TracingType_UUID),
```

Comment:
```suggestion
      tracingType: getObsFromEncounter(encounter, TracingType_UUID),
```

We should make this function performant that using a Map